examples/seismic/tti/tti_example.py

```diff
@@ -37,12 +37,14 @@ def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=250.0,
     # Define receiver geometry (spread across `x, y`` just below surface)
     rec, u, v, summary = solver.forward(save=save, autotune=autotune)
 
-    if preset == 'constant':
+    if preset == 'constant-tti':
         # With  a new m as Constant
         v0 = Constant(name="v", value=2.0, dtype=np.float32)
-        solver.forward(save=save, vp=v0)
+        solver.model.update('vp', v0.value)
+        solver.forward(save=save)
         # With a new vp as a scalar value
-        solver.forward(save=save, vp=2.0)
+        solver.model.update('vp', 2.0)
```

Comment:
perhaps the update should be performed internally to `forward`? 

if you had multiple calls to `forward` and each time you had to update one or more parameters "by hand", it would be painful

Reply:
Should probably yes be internal so the `dt` directly computed from runtime vo

Reply:
So, the same applies here: passing an scalar to `solver.forward() `would lead to an error
```
Traceback (most recent call last):
  File "devito/examples/seismic/tti/tti_example.py", line 103, in <module>
    run(shape=shape, spacing=spacing, nbl=args.nbl, tn=tn,
  File "devito/examples/seismic/tti/tti_example.py", line 48, in run
    solver.forward(vp=2., save=save)
  File "/home/oscar/devito/examples/seismic/tti/wavesolver.py", line 155, in forward
    summary = self.op_fwd(save).apply(src=src, rec=rec, u=u, v=v,
  File "/home/oscar/devito/devito/operator/operator.py", line 708, in apply
    args = self.arguments(**kwargs)
  File "/home/oscar/devito/devito/operator/operator.py", line 557, in arguments
    args = self._prepare_arguments(**kwargs)
  File "/home/oscar/devito/devito/operator/operator.py", line 449, in _prepare_arguments
    args.update(p._arg_values(**kwargs))
  File "/home/oscar/devito/devito/types/dense.py", line 843, in _arg_values
    for i, s in zip(self.dimensions, new.shape):
AttributeError: 'float' object has no attribute 'shape'
```
If you planned for this to be possible, then it is likely that there is some error somewhere else. If this is not the case, you need to use `model.update `before

Reply:
This looks like it ends up in the wrong spot (dense) instead of constant.

Reply:
So, are you saying you planned that the user could set a physical parameter i.e. velocity as a `Constant `or scalar  and pass it to the forward method of the `solver`, and it should work smoothly?